### DISCUSSION TITLE
windows issues

### DISCUSSION THREAD

**Discussion by roycw** [wg-easy/wg-easy#1143] [Q&A]
Hi guys simpley runing the docker example on my windows machine i get this 
~~~
| 2024-07-02T23:32:05.174Z Server Listening on http://0.0.0.0:51821
dev-wg-1  | 2024-07-02T23:32:05.188Z WireGuard Loading configuration...
dev-wg-1  | 2024-07-02T23:32:05.202Z WireGuard Configuration loaded.
dev-wg-1  | 2024-07-02T23:32:05.202Z WireGuard Config saving...
dev-wg-1  | 2024-07-02T23:32:05.204Z WireGuard Config saved.
dev-wg-1  | $ wg-quick down wg0
dev-wg-1  | $ wg-quick up wg0
dev-wg-1  | Error: WireGuard exited with the error: Cannot find device "wg0"
dev-wg-1  | This usually means that your host's kernel does not support WireGuard!
dev-wg-1  |     at /app/lib/WireGuard.js:64:19
dev-wg-1  |     at async /app/lib/WireGuard.js:62:9
~~

Can this run in a wondows env

**Comment by mitrios-hub** [comment]
[[> This usually means that your host's kernel does not support WireGuard!](https://github.com/wg-easy/wg-easy/issues/585)](https://github.com/wg-easy/wg-easy/issues/585)

**Comment by roycw** [comment]
Correct. but im having issue how to install it on windwos..... Can u shed some light ?

**Reply by offizium-berndstorath** [reply]
Get a terminal inside the container and run `wg-quick up wg0` manually to see the error

**Comment by pheiduck** [comment]
Well, Windows Docker uses WSL 2 backend maybe the wg module isn't enabled which is needed to get wireguard working.

**Reply by mitrios-hub** [reply]
I think it's better way to use hyper-v. I usually use it If I need to run Linux VM under Windows host.

**Comment by mitrios-hub** [comment]
> Hi guys simpley runing the docker example on my windows machine i get this
> 
> ```
> | 2024-07-02T23:32:05.174Z Server Listening on http://0.0.0.0:51821
> dev-wg-1  | 2024-07-02T23:32:05.188Z WireGuard Loading configuration...
> dev-wg-1  | 2024-07-02T23:32:05.202Z WireGuard Configuration loaded.
> dev-wg-1  | 2024-07-02T23:32:05.202Z WireGuard Config saving...
> dev-wg-1  | 2024-07-02T23:32:05.204Z WireGuard Config saved.
> dev-wg-1  | $ wg-quick down wg0
> dev-wg-1  | $ wg-quick up wg0
> dev-wg-1  | Error: WireGuard exited with the error: Cannot find device "wg0"
> dev-wg-1  | This usually means that your host's kernel does not support WireGuard!
> dev-wg-1  |     at /app/lib/WireGuard.js:64:19
> dev-wg-1  |     at async /app/lib/WireGuard.js:62:9
> ~~
> 
> Can this run in a wondows env
> ```

Why don't you use hyper-v?

**Reply by roycw** [reply]
i have other issues and i prefer to use the WSL-2 whihc is currentl enabled... how can i eable the wg module there ?

**Reply by mitrios-hub** [reply]
Wireguard works in the Linux kernel space, so it will not work in WSL.

Try the userspace implementation of wireguard - Boringtun (for example, a client based on it https://github.com/optman/wg-client)

**Comment by roycw** [comment]
ok, ill switch to hyper-v
I would think should be in the docs somewhere. windows is pretty common for containers. and i feel like tons of people have this issue.
Thanks for your help!
